### DISCUSSION TITLE
ignoreDeps not working for cimg/python

### DISCUSSION THREAD

**Discussion by tbrd** [renovatebot/renovate#10590] [Request Help]
I added python and cimg/python to ignoreDeps. Renovate is no longer reocmmending pythong upgrade to 3.9, but it is still suggesting cimg/python is upgraded. This requirement comes from a job in .circleci.

Is there any way to ignore this image without ignoring the entire circleci config?

```yaml
test-python:
    docker:
      - image: cimg/python:3.8
```

<img width="382" alt="Screenshot 2021-06-24 at 10 33 44" src="https://user-images.githubusercontent.com/102234/123239901-bb474780-d4d7-11eb-8ade-031503d8a84b.png">

```json
{
  "extends": [
    "config:base"
  ],
  "schedule": [
    "every weekend"
  ],
  "labels": [
    "dependencies"
  ],
  "separateMajorMinor": "false",
  "packageRules": [
    {
      "matchDepTypes": [
        "devDependencies"
      ],
      "matchUpdateTypes": [
        "major",
        "minor",
        "patch",
        "pin",
        "digest",
        "lockFileMaintenance",
        "rollback",
        "bump"
      ],
      "groupName": "devDependencies",
      "semanticCommitType": "chore",
      "automerge": true
    },
    {
      "matchDepTypes": [
        "dependencies"
      ],
      "matchUpdateTypes": [
        "major",
        "minor",
        "patch",
        "pin",
        "digest",
        "lockFileMaintenance",
        "rollback",
        "bump"
      ],
      "groupName": "dependencies",
      "semanticCommitType": "fix"
    }
  ],
  "ignoreDeps": ["python", "cimg/python"]
}
```

**Comment by rarkins** [comment]
What does the `packageFiles with updates` log contain for that dependency? We want to verify its `depName`.

**Reply by tbrd** [reply]
Thanks for the quick reply! I think I've answered this myself, but I'm not sure where to find that log - might be useful in the future?

**Reply by rarkins** [reply]
If you're running self-hosted, run with `LOG_LEVEL=debug` in your environment variables and search for whatever dependency/branch/PR that is causing the problem. If you are using the Renovate App, log into https://app.renovatebot.com/dashboard and locate the correct job log for when the problem occurred (e.g. when the PR was created).

**Reply by tbrd** [reply]
```json
{
            "depName": "cimg/python",
            "currentValue": "3.8",
            "replaceString": "cimg/python:3.8",
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "datasource": "docker",
            "depType": "docker",
            "versioning": "docker",
            "depIndex": 5,
            "warnings": [],
            "currentVersion": "3.8",
            "isSingleVersion": true,
            "fixedVersion": "3.8",
            "updates": [
              {
                "bucket": "latest",
                "newVersion": "3.9",
                "newValue": "3.9",
                "newMajor": 3,
                "newMinor": 9,
                "updateType": "minor",
                "branchName": "renovate/cimg-python-3.x"
              }
            ]
          },
          
```

**Reply by tbrd** [reply]
I've altered renovate.json to move the ignoredDeps to packageRules and it seems to have fixed it:
```
"packageRules": [
    {
      "matchPackageNames": ["python", "cimg/python"],
      "enabled": false
    },
    
    ```
    
    
<img width="494" alt="Screenshot 2021-06-24 at 11 03 14" src="https://user-images.githubusercontent.com/102234/123244418-d3b96100-d4db-11eb-9574-9634249bad04.png">

**Reply by tbrd** [reply]
This feels like a bug - I think it should also work in ignoreDeps

**Reply by rarkins** [reply]
If you can reproduce the `ignoreDeps` problem in a public repo then please raise it as a bug issue so that we can address it. Simpler the better

**Comment by tbrd** [comment]


**Reply by rarkins** [reply]
I don't think the `:3.8` part should be in the `depName` though. Or do you mean you accidentally added it in `ignoreDeps`? If Renovate is doing it then it sounds like a bug because `3.8` should be the `currentValue` and not part of `depName`.

**Reply by tbrd** [reply]
Oh! silly me, it didn't work at all. I just misread the generated PR
